03week/towersOfHanoi.js

```diff
@@ -19,23 +34,62 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
-
-}
-
-function isLegal() {
-  // Your code here
-
-}
+// variable to store the 3 valid entries
+const validEntry = ['a', 'b', 'c'];
 
-function checkForWin() {
-  // Your code here
+// function that verifies if the entries are true or false
+const isEntryValid = (startStack,endStack) => {
+  return validEntry.includes(startStack) && validEntry.includes(endStack);
+};
 
-}
 
 function towersOfHanoi(startStack, endStack) {
-  // Your code here
+  // two variables to store the stacks called by the user
+  const currentLocation = stacks[startStack];
+  const endLocation = stacks[endStack];
+
+  // function that determines move is valid and calls movePiece if true. 
+  function isLegal() {
+  // two variables. lastArrayItem stores the value called by the user in endStack
+  // removedItem stores the value called by the user in startStack
+    const lastArrayItem = endLocation[endLocation.length - 1];
+    const removedItem = currentLocation.pop();
+    
+    // function that pushes the removedItem onto the variable endLocation
+    function movePiece() {
+      endLocation.push(removedItem);
+      return stacks;
+    }
+  
+    if ((lastArrayItem > removedItem) || (endLocation.length < 1)) {
+      return movePiece();
+    } else {
+      currentLocation.push(removedItem);
+      console.log('Invalid Move');
+      return stacks;
+    }
+  }
+  
+  // function that puts all values in array a
+  const resetGame = () => {
+    const moveArray = endLocation.splice(0,4);
+    stacks['a'] = moveArray;
+  }
+  // function that runs after each movePiece and sees if all values are in stack c. Returns a string and resetGame();
+  const checkForWin = (endStack) => {
+    if (endStack == 'c'&& endLocation.length == 4) {
+      console.log('Winner Winner Chicken Dinner')
+      resetGame();
+    } 
+  }
+  
+
+  if (isEntryValid(startStack,endStack)) {
+    isLegal();
+    } else {
+     console.log ('not valid entry'); 
+    } 
+ checkForWin(endStack);
```

Comment:
you should only check for win if the entry is valid and is legal and after a move